packages/llamaindex/src/vector-store/azure/AzureAISearchVectorStore.ts

```diff
@@ -0,0 +1,1264 @@
+import {
+  AzureKeyCredential,
+  IndexDocumentsBatch,
+  KnownAnalyzerNames,
+  KnownSearchFieldDataType,
+  KnownVectorSearchAlgorithmKind,
+  KnownVectorSearchAlgorithmMetric,
+  KnownVectorSearchCompressionKind,
+  type LexicalAnalyzerName,
+  SearchClient,
+  type SearchField,
+  type SearchIndex,
+  SearchIndexClient,
+  type SearchOptions,
+  type SearchResult,
+  type SelectArray,
+  type SelectFields,
+  type SemanticConfiguration,
+  type SemanticSearch,
+  type SimpleField,
+  type VectorSearch,
+  type VectorSearchCompression,
+} from "@azure/search-documents";
+
+import { DefaultAzureCredential } from "@azure/identity";
+import { type BaseNode, MetadataMode } from "@llamaindex/core/schema";
+import { consoleLogger, getEnv } from "@llamaindex/env";
+import {
+  BaseVectorStore,
+  FilterCondition,
+  FilterOperator,
+  type MetadataFilters,
+  type VectorStoreBaseParams,
+  type VectorStoreQuery,
+  VectorStoreQueryMode,
+  type VectorStoreQueryResult,
+} from "../types.js";
+import { metadataDictToNode, nodeToMetadata } from "../utils.js";
+import {
+  AzureAISearchVectorStoreConfig,
+  type R,
+} from "./AzureAISearchVectorStoreConfig.js";
+import {
+  type AzureQueryResultSearchBase,
+  AzureQueryResultSearchDefault,
+  AzureQueryResultSearchHybrid,
+  AzureQueryResultSearchSemanticHybrid,
+  AzureQueryResultSearchSparse,
+} from "./AzureQueryResultSearch.js";
+
+/**
+ * Enumeration representing the supported index management operations
+ */
+export const enum IndexManagement {
```

Comment:
When accessing `IndexManagement.CREATE_IF_NOT_EXISTS` in userland, we get the following error:

```
Type error: Cannot access ambient const enums when 'isolatedModules' is enabled.
```

@himself65 any idea how to fix this?

Reply:
You canno export `const enum` it's the rule by typescript